relocation.bs

```diff
@@ -742,21 +742,40 @@ This signature was picked as it completes the C++ tripartite value system.
 The copy constructor creates a new instance from an lvalue, the move constructor 
 from an xvalue, and then the relocation constructor from a prvalue.
 
-Despite what its signature suggests, the constructor does not take a copy of the
-source object as parameter. In fact, the constructor benefits from a special 
-aliasing to ensure that the address of its parameter is the same as the source
-object that is being relocated (as if the parameter was captured by reference).
-It is fine for the relocation constructor to provide such aliasing as constructors
-are special functions, which we cannot take the address of.
+Note: a further benefit of this syntax is that it is currently ill-formed, and
+thus available for extension.
+
+A point of confusion may be that the syntax implies an infinite regression: the
+parameter must be constructed, which requires a prior call to the relocation
+constructor, and so on. This is not the case; if the source object was previously
+a glvalue the operand of the `reloc` operator, it was transformed into a prvalue
+immediately before entering the relocation constructor, and the parameter of the
+relocation constructor is that same prvalue. (If the source object was already a
+prvalue, there is no issue; the parameter is that prvalue.)
+
+An attractive intuition is that the parameter aliases the source object in the
+same way as a reference or a structured binding declaration.
+However, this is misleading; the lifetime of a source object glvalue has already
```

Comment:
Why is that exactly? Shouldn't it end when the relocation ctor returns?

Reply:
To accord with https://wg21.link/basic.life#1.4

> ... The lifetime of an object o of type T ends when: ... if T is a class type, the destructor call starts, or ...

Worth adding a note, though, since not everyone is aware of this.